### PR TITLE
simplify cmpBuf

### PR DESCRIPTION
**Pull request by MaxGraey**

### CODE REVIEW

File: assembly/util.ts [hunk 1]
```diff
@@ -28,26 +28,19 @@ export function padBuf(buf: Uint8Array, length: usize): Uint8Array {
...
-    if (a == b) {
-      continue
-    } else if (a < b) {
-      res = -1
-    } else {
-      res = 1
-    }
-    // Break after first un-equal byte encountered
-    break
+    let a = unchecked(buf[i])
```
Comment (s1na) [inline assembly/util.ts]: For my information, is the purpose of `unchecked` to avoid unnecessary bound checks, when the bounds are known?

Comment (MaxGraey) [reply assembly/util.ts]: yes, when you could safely guarantee that checks unnecessary. In future this manual optimization will not need.

**Comment by s1na** [comment]
Hey @MaxGraey, we don't have much experience with AS and value your expertise there. If you ever wanted to make further changes, I've added you as a collaborator to make things easier.

**Comment by MaxGraey** [comment]
Thanks! I'm just gonna make things better from time to time. I don't think that requires collaborator's role)

**Pull request merged**